pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/FunctionsImpl.java

```diff
@@ -595,16 +594,23 @@ public Response triggerFunction(final String tenant, final String namespace, fin
         String inputTopicToWrite;
         if (topic != null) {
             inputTopicToWrite = topic;
+        } else if (functionMetaData.getFunctionDetails().getSource().getTopicsToSerDeClassNameMap().size() == 1) {
```

Comment:
I think the right way of doing this would be to convert the old format into new format. Thus when you receive and decode a functiondetails proto, add a normalize method of some sort that puts all old style stuff into new style.
The reason is because there are other pieces of code which will rely on data being present only in inputSpecs.

Reply:
changed to normalizing the function metadata